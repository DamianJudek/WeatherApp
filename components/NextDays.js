import React from 'react';
import { View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import Day from './Day';

const styles = StyleSheet.create({
  container: {
    flex: 2,
    flexDirection: 'row',
    justifyContent: 'center',
    borderRadius: 50,
    padding: 10,
    backgroundColor: 'rgba(255, 255, 255, 0.2)',
  },
  section: {
    flex: 1,
    borderRightWidth: 1,
    borderColor: '#fff',
    justifyContent: 'center',
    paddingHorizontal: '2%',
  },
  lastSection: {
    borderRightWidth: 0,
  },
});

class NextDays extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  static propTypes = {
    days: PropTypes.array,
  };

  render() {
    const [first, second, third, fourth] = this.props.days;
    return (
      <View style={styles.container}>
        <Day style={styles.section} weather={first} />
        <Day style={styles.section} weather={second} />
        <Day style={styles.section} weather={third} />
        <Day style={[styles.section, styles.lastSection]} weather={fourth} />
      </View>
    );
  }
}

export default NextDays;
