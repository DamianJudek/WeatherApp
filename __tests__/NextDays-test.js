import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NextDays from '../components/NextDays';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders correctly', () => {
  const snap = renderer
    .create(
      <NextDays
        days={[
          {
            iconDay: '10d',
            iconNight: '04n',
            maxTemp: 18,
            minTemp: 11,
            name: 'Mon.',
          },
          {
            iconDay: '04d',
            iconNight: '04n',
            maxTemp: 17,
            minTemp: 11,
            name: 'Tue.',
          },
          {
            iconDay: '01d',
            iconNight: '01n',
            maxTemp: 17,
            minTemp: 10,
            name: 'Wed.',
          },
          {
            iconDay: '04d',
            iconNight: '01n',
            maxTemp: 19,
            minTemp: 11,
            name: 'Thu.',
          },
        ]}
      />,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
