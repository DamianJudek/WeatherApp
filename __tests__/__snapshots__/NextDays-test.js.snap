// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.3)",
      "borderRadius": 50,
      "flex": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 0,
      "padding": 10,
      "transform": Array [
        Object {
          "scale": 0.8,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#fff",
        "borderRightWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "paddingHorizontal": "2%",
        "transform": Array [
          Object {
            "translateY": 35,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Mon.
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "http://openweathermap.org/img/wn/10d@2x.png",
          }
        }
        style={
          Array [
            Object {
              "height": 55,
              "width": 55,
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      >
        18℃
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(255, 255, 255, 0.2)",
            "borderTopWidth": 1,
            "height": 0,
            "transform": Array [
              Object {
                "rotate": "-45deg",
              },
            ],
            "width": "100%",
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "flex-end",
            },
          ]
        }
      >
        11℃
      </Text>
      <View>
        <Image
          resizeMode="contain"
          source={
            Object {
              "uri": "http://openweathermap.org/img/wn/04n@2x.png",
            }
          }
          style={
            Array [
              Object {
                "height": 55,
                "width": 55,
              },
              Object {
                "alignSelf": "flex-end",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "#fff",
        "borderRightWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "paddingHorizontal": "2%",
        "transform": Array [
          Object {
            "translateY": 35,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Tue.
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "http://openweathermap.org/img/wn/04d@2x.png",
          }
        }
        style={
          Array [
            Object {
              "height": 55,
              "width": 55,
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      >
        17℃
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(255, 255, 255, 0.2)",
            "borderTopWidth": 1,
            "height": 0,
            "transform": Array [
              Object {
                "rotate": "-45deg",
              },
            ],
            "width": "100%",
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "flex-end",
            },
          ]
        }
      >
        11℃
      </Text>
      <View>
        <Image
          resizeMode="contain"
          source={
            Object {
              "uri": "http://openweathermap.org/img/wn/04n@2x.png",
            }
          }
          style={
            Array [
              Object {
                "height": 55,
                "width": 55,
              },
              Object {
                "alignSelf": "flex-end",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "#fff",
        "borderRightWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "paddingHorizontal": "2%",
        "transform": Array [
          Object {
            "translateY": 35,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Wed.
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "http://openweathermap.org/img/wn/01d@2x.png",
          }
        }
        style={
          Array [
            Object {
              "height": 55,
              "width": 55,
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      >
        17℃
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(255, 255, 255, 0.2)",
            "borderTopWidth": 1,
            "height": 0,
            "transform": Array [
              Object {
                "rotate": "-45deg",
              },
            ],
            "width": "100%",
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "flex-end",
            },
          ]
        }
      >
        10℃
      </Text>
      <View>
        <Image
          resizeMode="contain"
          source={
            Object {
              "uri": "http://openweathermap.org/img/wn/01n@2x.png",
            }
          }
          style={
            Array [
              Object {
                "height": 55,
                "width": 55,
              },
              Object {
                "alignSelf": "flex-end",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "#fff",
        "borderRightWidth": 0,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "paddingHorizontal": "2%",
        "transform": Array [
          Object {
            "translateY": 35,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Thu.
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "http://openweathermap.org/img/wn/04d@2x.png",
          }
        }
        style={
          Array [
            Object {
              "height": 55,
              "width": 55,
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      >
        19℃
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(255, 255, 255, 0.2)",
            "borderTopWidth": 1,
            "height": 0,
            "transform": Array [
              Object {
                "rotate": "-45deg",
              },
            ],
            "width": "100%",
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "flex-end",
            },
          ]
        }
      >
        11℃
      </Text>
      <View>
        <Image
          resizeMode="contain"
          source={
            Object {
              "uri": "http://openweathermap.org/img/wn/01n@2x.png",
            }
          }
          style={
            Array [
              Object {
                "height": 55,
                "width": 55,
              },
              Object {
                "alignSelf": "flex-end",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
